src/scancode/format.py

```diff
@@ -105,14 +106,21 @@ def create_html_app_assets(output_file):
     HtmlAppAssetCopyError if the copy was not possible.
     """
     try:
+        if is_stdout(output_file):
+            raise HtmlAppAssetCopyWarning()
         assets_dir = join(get_template_dir('html-app'), 'assets')
+
         tgt_dirs = get_html_app_files_dirs(output_file)
-        if not tgt_dirs:
-            raise HtmlAppAssetCopyWarning()
         target_dir = join(*tgt_dirs)
         if exists(target_dir):
             fileutils.delete(target_dir)
         fileutils.copytree(assets_dir, target_dir)
+
+        # write json data
+        import json
+        root_path, assets_dir = get_html_app_files_dirs(output_file)
+        with open(join(root_path, assets_dir, 'data.json'), 'w') as f:
+            f.write('data=' + json.dumps(results))
```

Comment:
should the variable be called scandata instead ... data feels quite generic otherwise?